src/auth/ejabberd_auth.erl

```diff
@@ -578,3 +583,13 @@ fold_auth_modules([AuthModule | AuthModules], F, CurAcc) ->
         {stop, Value} ->
             Value
     end.
+
+is_user_limit_per_domain_exceeded(Domain) ->
```

Comment:
Function name is misleading. If there are exactly as many users as the limit, it is **not** exceeded, yet it returns `true`.